airbyte-api/src/main/openapi/config.yaml

```diff
@@ -3769,6 +3769,9 @@ components:
             $ref: "#/components/schemas/JobConfigType"
         configId:
           type: string
+        includingJobId:
+          description: If the job with this ID exists for the specified connection, returns all jobs created after and including this job, or the full pagination pagesize if that list is smaller than a page. Otherwise, this field is ignored.
```

Comment:
This feels hard to use from a client perspective.
If I call the API using this attribute, the only way to know that the jobId doesn't exist is to check all the results to see if the jobId appears.
I am not sure about your use cases, but returning an empty list if the job doesn't exists feels more straightforward.

Reply:
The use case is that someone can link to a particular job by creating a url such as https://cloud.airbyte.io/workspaces/a385d3d9-4512-4a3a-9240-cea3d237c822/connections/aa3b8ef1-2f39-48b8-bad4-dd12b0d3876d/status#575674::0 -- notice the `#575674::0` at the end; the first number is the job ID (second is the attempt ID). This type of link can also be generated by clicking the "link" icon button on a job in the UI.

My reasoning here was that if someone correctly supplies a job ID that is in the connection, this will work as expected. Otherwise, they are specifying a job ID that doesn't exist in the connection, so there is nothing for us to do with that `#jobId::attemptId` suffix, and we should therefore just ignore that and return the normal job list. 

However, your comment has me thinking that it might be a better user experience if we instead showed a specific message indicating that the linked job could not be found, with a button to return them to the normal job list page. With that approach, I think your suggestion of returning an empty list of the job cannot be found makes sense, so I can look into making that change

Reply:
I've changed this to return an empty list if the target job cannot be found

Reply:
Sorry for the late reply. Can I voice the opposite opinion here? :D

I feel if we can't find the ID that's passed in, we should just return the specified page size instead and ignore trying to find that job. I think that would play nicer together with the frontend, since we otherwise might want to add additional logic in the FE.

The ID will just be part of the fragment of the URL, a user might accidentally change that to an invalid ID. If you try to load the page like that, we'd now just see an empty page with a "Load more" button. That feels like a bit weird behavior to me, and I think we rather want to just load the 25 first jobs still in this case. We can of course address that in the FE by making another API call if we see the list is empty, but had a totalCount that's > 0, but I feel it would be easier specifying this API in a way that it will return the regular page size if the ID isn't found.

Reply:
@timroes I have actually already put up FE changes to show a message like `The linked job could not be found` with a link that takes them back to the normal sync history page: https://github.com/airbytehq/airbyte/pull/16517
It didn't end up being too much more complex to do this

Reply:
Even better UX. Ignore my comment then